osu.Game/Skinning/GameplaySkinComponentLookup.cs

```diff
@@ -25,7 +27,41 @@ public GameplaySkinComponentLookup(T component)
             Component = component;
         }
 
-        protected virtual string RulesetPrefix => string.Empty;
-        protected virtual string ComponentName => Component.ToString();
+        public virtual RulesetInfo? Ruleset => null;
+
+        public override string ToString()
+        {
+            if (Ruleset == null)
+                return Component.ToString();
+
+            return $"{Component.ToString()} (\"{Ruleset.Name}\" only)";
+        }
```

Comment:
What's this `ToString()` used by? I _very_ much don't like display logic depending on these, because then innocent changes elsewhere change display...

If anything display depends on this I'd hope this can be moved to something that looks like display.

Reply:
This is for display in the skin editor dropdown. I'll try moving it to a custom dropdown implementation (the dropdown is using `ISkinComponent` directly as its items).